app/models/package_manager/maven.rb

```diff
@@ -17,27 +18,52 @@ class Maven < Base
       "http://www.eclipse.org/org/documents/edl-v10" => "Eclipse Distribution License (EDL), Version 1.0",
     }.freeze
 
+    PROVIDER_MAP = {
+      "SpringLibs" => SpringLibs,
+      "Maven" => MavenCentral,
+    }.freeze
+
     def self.package_link(project, version = nil)
+      if version
+        db_version = project.versions.find_by(number: version)
+        unless db_version&.repository_sources.blank?
+          repository_source = db_version.repository_sources.first
+          return PROVIDER_MAP[repository_source].package_link(project, version)
+        end
+      end
+
       MavenUrl.from_name(project.name, repository_base).search(version)
     end
 
     def self.download_url(name, version = nil)
+      if version
+        project = Project.find_by(name: name, platform: "Maven")
+        db_version = project.versions.find_by(number: version)
+        unless db_version&.repository_sources.blank?
+          repository_source = db_version.repository_sources.first
```

Comment:
Can't we just sort of default these now?
`repository_source = db_version.repository_sources&.first.presence? || "Maven"` kinda thing?

Reply:
That might help make it more explicit. I wrote each of these to fail down to what was already here if there wasn't a repository source for the version, which should be the same as saying `|| "Maven"`.

Reply:
Or maybe even a default constant up top? DEFAULT_SOURCE = "Maven" and then just use that all over

Reply:
I've made "MavenCentral" the default now which simplifies the logic. Pick the first repository source if it's there and if not just use the MavenCentral provider.

Reply:
Awesome, looks good! 👍